src/routes/Apps/App/TrainDialogs.tsx

```diff
@@ -922,13 +951,18 @@ class TrainDialogs extends React.Component<Props, ComponentState> {
 
     @OF.autobind
     async openTrainDialog(trainDialog: CLM.TrainDialog, roundIndex: number, scoreIndex: number | null) {
-
         const selectedActivityIndex = DialogUtils.activityIndexFromRound(trainDialog, roundIndex, scoreIndex) || null
-        await this.onClickTrainDialogItem(trainDialog, EditDialogType.TRAIN_ORIGINAL, selectedActivityIndex)
+        await this.selectTrainDialog(trainDialog, EditDialogType.TRAIN_ORIGINAL, selectedActivityIndex)
     }
 
     @OF.autobind
-    async onClickTrainDialogItem(trainDialog: CLM.TrainDialog, editType: EditDialogType = EditDialogType.TRAIN_ORIGINAL, selectedActivityIndex: number | null = null) {
+    async onClickTrainDialogItem(trainDialog: CLM.TrainDialog) {
+        const { history } = this.props
+        let url = `/home/${this.props.app.appId}/trainDialogs?id=${trainDialog.trainDialogId}`
+        history.push(url, { app: this.props.app })
+    }
+
+    async selectTrainDialog(trainDialog: CLM.TrainDialog, editType: EditDialogType = EditDialogType.TRAIN_ORIGINAL, selectedActivityIndex: number | null = null) {
```

Comment:
Don't feel too strongly but was wondering if there is better name `selectTrainDialog` ?

From what I can tell there are cases where we open the model even if the user didn't click on an item so onClick is no longer accurate and it makes sense to split these functions but they still seem to overlap. Select is very much like Click

Is the effect of this function showing a dialog in the modal?  Was thinking `viewTrainDialog` but then it sounds render related.

Reply:
openLogDialog?

Reply:
Ok, I think that more clearly describes effect of function.  Also, not sure if it has effect, but you can make it `private` to further tell that the component should be using `onClickTrainDialogItem` in the JSX but `selectTrainDialog` internally.